misc/tests/conftest.py

```diff
@@ -0,0 +1,8 @@
+def pytest_addoption(parser):
+    parser.addoption("--suppress", action="append", default=[],
+                     help="Suppression files used in test-memory-leaks.py. These files must be located in the folder valgrind.")
+
+def pytest_generate_tests(metafunc):
+    if "suppression_files" in metafunc.fixturenames:
+        tags = list(metafunc.config.option.suppress)
```

Comment:
`tags` is an unclear name. The way I understand it, this variable holds the list of suppression files, so I'd call it `suppression_files`.